sheets_to_JSON_genotype.R

```diff
@@ -0,0 +1,121 @@
+# library packages
+library(googlesheets4)
+library(dplyr)
+library(tidyr)
+library(jsonlite)
+
+
+
+
+# read in the data
+url <- "https://docs.google.com/spreadsheets/d/1lwVMGT-TQaWbMWvi3hdqWuEthZvaKGOImINAqXguPaM"
+meta <- read_sheet(url, sheet="Tables")
+meta_tsv <- meta %>%
+  mutate(entity="meta") %>%
+  select(entity, required=Required, table=Table)
+
+
+
+
+# pull tables that will be converted to JSON
+table_names <- meta_tsv$table
+tables <- lapply(table_names, function(x){read_sheet(url, sheet = x, col_types = "c")})
+names(tables) <- table_names
+rm(list = c("table_names", "url"))
+
+
+
+
+# rename things as shown in the toy example
+for (i in 1:length(tables)) {
+  names(tables[[i]]) <- c("required", "name", "description", "references", "data_type", "enumerations", "examples", "notes_comments")
```

Comment:
This will only work as long as the order of columns in the google sheet stays the same. To make it more robust, you could explicitly map old to new names: "Required" -> "required" (or just use `tolower()`), etc. That would also let you reuse this bit of code for the other data model conversion scripts (sheets_to_JSON_gsr, etc), since their google sheets don't have exactly the same column order as this one.

Reply:
Thanks for this suggestion, Stephanie! I will use the `tolower()` function to keep the code generalizable. I also had some preferance for the order of the JSON keys (e.g. it makes sense to have the table name come before identifying if it is required), however the optimal order can be changed directly in the Google Sheets file, as needed.